File: 백준/Silver/1926. 그림/그림.py
import sys
input=sys.stdin.readline

n,m=map(int,input().split())
graph=[list(map(int,input().split())) for _ in range(n)]

dx=[1,-1,0,0]
dy=[0,0,1,-1]
result=[]

def bfs(start,end):
    q=[[start,end]]
    graph[start][end]=0
    count=0
    while q:
        x,y=q.pop(0)
        count+=1
        for i in range(4):
            nx=x+dx[i]
            ny=y+dy[i]
            if 0<=nx<n and 0<=ny<m and graph[nx][ny]==1:
                q.append([nx,ny])
                graph[nx][ny]=0
    result.append(count)

for i in range(n):
    for j in range(m):
        if graph[i][j]==1:
            bfs(i,j)
print(len(result))
print(max(result) if len(result)!=0 else 0)

File: 백준/Silver/1926. 그림/그림.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
	static int graph[][];
	static int n,m;
	static int mx=0,cnt=0;
	static int[] dx = {-1,1,0,0};
	static int[] dy= {0,0,-1,1};
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine()," ");
		n=Integer.parseInt(st.nextToken());
		m=Integer.parseInt(st.nextToken());
		
		graph=new int[n][m];
		
		for(int i=0;i<n;i++) {
			st=new StringTokenizer(br.readLine()," ");
			for(int j=0;j<m;j++) {
				graph[i][j]=Integer.parseInt(st.nextToken());
			}
		}
		
		for(int i=0;i<n;i++) {
			for(int j=0;j<m;j++) {
				if(graph[i][j]==1) {
					bfs(i,j);
				}
			}
		}
		System.out.println(cnt);
		System.out.println(mx);
	}
	
	static void bfs(int x,int y) {
		int count=1;
		cnt++;
		Queue<int[]> q = new ArrayDeque<>();
		q.add(new int[] {x,y});
		graph[x][y]=0;
		while(!q.isEmpty()) {
			int a[]=q.poll();
			x=a[0];
			y=a[1];
			for(int i=0;i<4;i++) {
				int nx=x+dx[i];
				int ny=y+dy[i];
				if(nx>=0 && nx<n && ny>=0 && ny<m && graph[nx][ny]==1) {
					q.add(new int[] {nx,ny});
					graph[nx][ny]=0;
					count++;
				}
			}
		}
		mx=Math.max(mx, count);
	}
}